ide/notifications.linux/src/org/netbeans/modules/notifications/linux/LinuxNotificationDisplayer.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.netbeans.modules.notifications.linux;
+
+import com.sun.jna.Native;
+import com.sun.jna.Pointer;
+import java.awt.event.ActionListener;
+import java.util.Optional;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.swing.Icon;
+import javax.swing.JComponent;
+import org.openide.awt.Notification;
+import org.openide.awt.NotificationDisplayer;
+import org.openide.util.Lookup;
+import org.openide.util.lookup.ServiceProvider;
+import org.netbeans.modules.notifications.linux.jna.Libnotify;
+
+/**
+ *
+ * @author Hector Espert
+ */
+@ServiceProvider(service = LinuxNotificationDisplayer.class, position = 50)
+public class LinuxNotificationDisplayer extends NotificationDisplayer {
+
+    private static final Logger LOG = Logger.getLogger(LinuxNotificationDisplayer.class.getName());
+
+    private static final String LIBNOTIFY = "libnotify.so.4";
+
+    private static final String APP_NAME = "netbeans";
+
+    private Optional<Libnotify> optionalLibnotify;
+
+    public LinuxNotificationDisplayer() {
+        try {
+            this.optionalLibnotify = Optional.of(Native.load(LIBNOTIFY, Libnotify.class));
```

Comment:
I know we typically do not use module installer these days, but I think putting the loading of the native library in a module install to let the install fail if the library is not available may be worth a try - with this I can imagine all other if / else / optional  stuff may not be needed anymore.

Reply:
@svenreimers Is there an example of ModuleInstall that I can use?